src/CodeGen_Hexagon.cpp

```diff
@@ -100,6 +101,11 @@ void CodeGen_Hexagon::compile_func(const LoweredFunc &f,
 
     Stmt body = f.body;
 
+    debug(1) << "Aligning loads for HVX....\n";
+    body = align_loads(body, target);
+    body = simplify(body);
```

Comment:
I think this might as well also run CSE now too (prior to simplification?). The bulk of the work eliminating loads will come from store forwarding, but CSE should still at least eliminate some of the loads from double wide loads that got lowered to 4 aligned loads (which would become 3 after CSE?). Maybe CSE will even hit a bunch of the loads from stencils that get aligned?